lib/media/streaming_engine.js

```diff
@@ -1743,36 +1744,104 @@ shaka.media.StreamingEngine = class {
     const repId = this.mediaStates_.get(ContentType.VIDEO)
         .stream.originalId.split(',').pop();
     if (repId == null) {
-      shaka.log.warning('getPresentationLatency cannot get video repId');
+      shaka.log.warning('getProducerReferenceTimeInfo cannot get video repId');
       return null;
     }
     if (data.periodIdMap == null) {
-      shaka.log.warning('getPresentationLatency cannot get periodIdMap');
+      shaka.log.warning('getProducerReferenceTimeInfo cannot get periodIdMap');
       return null;
     }
     const periodId = this.getPeriodIdFromPresentationTime(pt, data.periodIdMap);
-    const rv = [];
     const keyPrefix = `${periodId}_${repId}`;
+    const prtData = [];
     for (const [key, producerReferenceTime] of data.prtMap.entries()) {
       if (!key.startsWith(keyPrefix)) {
         continue;
       }
       const wca = producerReferenceTime.wca;
       const pta = producerReferenceTime.pta;
       const type = producerReferenceTime.type;
-      const latency = Math.round((wc - wca) - (pt - pta));
-      shaka.log.debug(`computing presentation latency for
+      const wcaPrft = producerReferenceTime.wca_prft;
+      const ptaPrft = producerReferenceTime.pta_prft;
+      const hasPrft = wcaPrft >= 0 && ptaPrft >= 0;
+
+      shaka.log.debug(`Getting ProducerReferenceTimeInfo for
           representation ${repId}, 
           wc=${wc},
           wcOffset=${clockOffset},
-          wca=${wca}, 
-          pt=${pt} 
-          pta=${pta}
-          type=${type}
-          latency=${latency}ms`);
-      rv.push({type, latency});
-    }
-    return rv;
+          wca=${wca},
+          wca from PRFT=${wcaPrft},
+          pt=${pt},
+          pta=${pta},
+          pta from PRFT=${ptaPrft},
+          type=${type}`);
+
+      prtData.push({
+        type,
+        wca: hasPrft ? wcaPrft : wca,
+        pta: hasPrft ? ptaPrft : pta,
+        inband: hasPrft,
+      });
+    }
+    return prtData;
+  }
+
+  /**
+   * @suppress {missingProperties}
+   * Get presentation latency
+   * If encoder latency is not available, fallback to captured latency
+   *
+   * Used in :
+   * - Web Player SDK - Debug overlay
+   * - SimpleLLAbrManager::increaseVideoBitrate
+   * - LiveCatchUpController::getLatency
+   *
+   * @return {Object|null}
+   */
+  getPresentationLatency() {
```

Comment:
Consider renaming it as getPresentationLatencyInfo if the return type is not a simple numeric latency value